### ISSUE TITLE
Component should accept className and styles should all be obj OR string

### ISSUE THREAD

**Issue by L5eoneill** [AllenFang/react-bootstrap-table#647]
Container style, all the other styles... need to let custom css module style names go in there in place of style objects. Also, the main component should accept standard Bootstrap 'className' prop.

**Comment by AllenFang** [comment]
@L5eoneill, all the className accept the string or a function which generate class name dynamically. and the style only accept object which follow jsx principle

```js
<BootstrapTable insertRow exportCSV data={ products }
        tableStyle={ { border: '#0000FF 2.5px solid' } }
        containerStyle={ { border: '#FFBB73 2.5px solid' } }
        trClassName='tr-string-example'
>
          <TableHeaderColumn dataField='id' isKey={ true }>Product ID</TableHeaderColumn>
          <TableHeaderColumn dataField='name'>Product Name</TableHeaderColumn>
          <TableHeaderColumn dataField='price'>Product Price</TableHeaderColumn>
      </BootstrapTable>
```

BTW, I'm not sure if I've answered your question, feel free to discuss with me :)
thanks

**Comment by L5eoneill** [comment]
Yeah, doesn't completely address the issue, which is that many of the elements your component creates don't receive class names. Accepting a style object just isn't good enough. We get:

react-bs-table-container
react-bs-table
react-bs-container-header
react-bs-container-body

From your docs, none of these can take on a custom css class name, so if you're using css-modules, you have to work really hard to override even the smallest of default bootstrap stylings, using tricks like !important and :global(.react-bs-table) and such, which are frowned upon.

**Comment by AllenFang** [comment]
@L5eoneill, ok,  currently, ```tableHeaderClass``` and ```tableBodyClass``` only apply class name to header and body ```table```. Maybe  I can support another props for supporting to apply class name to following elements:

react-bs-table-container
react-bs-table
react-bs-container-header
react-bs-container-body

**Comment by L5eoneill** [comment]
Yes, that'd be excellent. Thanks.

**Comment by AllenFang** [comment]
use ```containerClass``` to apply class to ```react-bs-table-container```
use ```tableContainerClass``` to apply class to ```react-bs-table```
use ```headerContainerClass``` to apply class to ```react-bs-container-header```
use ```bodyContainerClass``` to apply class to ```react-bs-container-body```

Released on ```v2.4.4```

**Comment by AllenFang** [comment]
there's one thing to notice, on v2.4.4 I move the css files from css to dist folder, remember to update the css path. I apologize for it.
